src/Riok.Mapperly/Descriptors/Mappings/ExistingTarget/ForEachSetDictionaryExistingTargetMapping.cs

```diff
@@ -12,25 +12,40 @@ namespace Riok.Mapperly.Descriptors.Mappings.ExistingTarget;
 public class ForEachSetDictionaryExistingTargetMapping : ExistingTargetMapping
 {
     private const string LoopItemVariableName = "item";
+    private const string ExplicitCastVariableName = "targetDict";
     private const string KeyValueKeyPropertyName = nameof(KeyValuePair<object, object>.Key);
     private const string KeyValueValuePropertyName = nameof(KeyValuePair<object, object>.Value);
 
     private readonly ITypeMapping _keyMapping;
     private readonly ITypeMapping _valueMapping;
+    private readonly INamedTypeSymbol? _explicitCast;
 
     public ForEachSetDictionaryExistingTargetMapping(
         ITypeSymbol sourceType,
         ITypeSymbol targetType,
         ITypeMapping keyMapping,
-        ITypeMapping valueMapping)
+        ITypeMapping valueMapping,
+        INamedTypeSymbol? explicitCast)
         : base(sourceType, targetType)
     {
         _keyMapping = keyMapping;
         _valueMapping = valueMapping;
+        _explicitCast = explicitCast;
     }
 
     public override IEnumerable<StatementSyntax> Build(TypeMappingBuildContext ctx, ExpressionSyntax target)
     {
+        if (_explicitCast is not null)
+        {
+            var type = FullyQualifiedIdentifier(_explicitCast);
```

Comment:
Could you add a comment how the generated code looks like to make it faster to understand, we started that a while ago and it makes the mappings a lot easier to read.